### PR TITLE
Benja merge cleanup 971

### PR DESCRIPTION
**Pull request by Shriukan33**
This PR does a cleanup of the #997 PR.

Mostly, it's improvements on responsiveness and code quality on front-end, and a minor change in the backend to use a non GUI backend for matplotlib.

### CODE REVIEW

**Comment by Shriukan33** [comment]
Actually there is a line that was deleted and that is important for flag feature, I'm re-adding it as well as adding a selenium test to catch it in the future.

**Comment by Shriukan33** [comment]
Added the aforementioned test !

File: transport_nantes/mobilito/templates/mobilito/mobilito_session_summary.html [hunk 1]
```diff
@@ -48,133 +48,131 @@
...
 
-        {% with start=mobilito_session.start_timestamp end=mobilito_session.end_timestamp %}
-                <div class="dropdown show">
-                    <a href="#" role="button" id="dropdownMenuLink" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
-                        <i class="fa-solid fa-ellipsis ml-md-3"></i>
-                    </a>
-                    <div class="dropdown-menu dropdown-menu-right" aria-labelledby="dropdownMenuLink">
-                        <a class="dropdown-item" href="#"
-                        {% if user_has_reported_this_session or request.user == mobilito_session.user.user %}
-                            style="pointer-events: none; color: grey;"
```
Comment (JeffAbrahamson) [inline transport_nantes/mobilito/templates/mobilito/mobilito_session_summary.html:108]: Would this be better as mobilitain-grey?

**Review by JeffAbrahamson — APPROVED** [review]
LGTM.

Comments, for future commits:
* the flag doesn't display for me on beta
* the map is rather tall and, as the screen shrinks, stays that tall, meaning on mobile it takes quite a lot of space even though it's not the main point

File: transport_nantes/mobilito/templates/mobilito/mobilito_session_summary.html (hunk 1 above)
Comment (Shriukan33) [reply transport_nantes/mobilito/templates/mobilito/mobilito_session_summary.html:108]: This is just the font color when one has already flagged a session or is unable to. I don't believe it warrants a special color other than the internet wide grey for disabled stuff :slightly_smiling_face:

**Comment by Shriukan33** [comment]
> LGTM.
> 
> Comments, for future commits:
> 
>     * the flag doesn't display for me on beta


It's still an ellipsis for now, and it does display but it's easily miss-able : 
image

I think the flag icon will be more noticeable while the color scheme is quite discreet, need to find a balance here.

> 
>     * the map is rather tall and, as the screen shrinks, stays that tall, meaning on mobile it takes quite a lot of space even though it's not the main point

I can limit its height, it didn't hit me as "too big" to be honest but maybe it is. I mean, I can understand you may find it too big.

**Comment by JeffAbrahamson** [comment]
Tracking map size at #1013, flag at #1027.

**Comment by Shriukan33** [comment]
Thanks for opening issues ! :heart_eyes:

**Pull request merged**